### ISSUE TITLE
Make hook timeout configurable

### ISSUE THREAD

**Issue by thomastaylor312** [kubernetes/helm#1678]
Currently there is no way to set a custom timeout value for lifecycle hooks. Because `Jobs` just restart if they fail with a non-zero exit code (no matter the restart policy), it would be nice to have a configurable timeout value for how long to wait on lifecycle hooks.

**Comment by thomastaylor312** [comment]
If I get some time, I might be able to open a PR for this

**Comment by technosophos** [comment]
That would be cool. Where would the timeout information go?

**Comment by thomastaylor312** [comment]
I would think it would be a command line flag and it would just get passed to tiller.

**Comment by technosophos** [comment]
So it would be timeout per install? That could work quite well.

**Comment by thomastaylor312** [comment]
Yeah definitely per install/upgrade/rollout is what I was thinking
